src/Jsonyte/Serialization/Metadata/ResourceObjectTypePropertyInfo.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Jsonyte.Serialization.Attributes;
+
+namespace Jsonyte.Serialization.Metadata
+{
+    /// <summary>
+    /// This contains the information to serialize the Type attribute when it is declared using the ResourceObjectAttribute
+    /// </summary>
+    internal sealed class ResourceObjectAttributeMemberInfo: JsonMemberInfo<string>
+    {
+        private ResourceObjectAttribute resourceObject;
+
+        public ResourceObjectAttributeMemberInfo(ResourceObjectAttribute resourceObject, JsonSerializerOptions options)
+            : base(typeof(ResourceObjectAttribute).GetProperty(nameof(resourceObject.Type))!, typeof(string), JsonIgnoreCondition.Never, options.GetConverter(typeof(string)), options)
```

Comment:
Do we even need some of these params, given this is a 'fake' JSON member? Doesn't save on compute time, since this is expensive setup done once for each type encountered, but it might clean up this code a bit?

Reply:
Because this is a 'fake' member and assumes that the 'Type' property doesn't exist, I'm not sure we need to worry about reflecting the type property in the base constructor call. Maybe you can just pass null to the base ctor and have it work? I might be wrong though and the unit tests will complain...

Reply:
It looks like the unit tests do fail and we get a warning about passing null to a non-nullable parameter.